src/main/scala/dpla/ingestion3/harvesters/api/MdlHarvester.scala

```diff
@@ -33,47 +33,40 @@ class MdlHarvester(spark: SparkSession,
   override def localHarvest: DataFrame = {
     implicit val formats = DefaultFormats
 
-    // Mutable vars for controlling harvest loop
-    var continueHarvest = true
-    var requestUrl = getFirstUrl(queryParams)
+    // Mutable var for controlling harvest loop
+    var requestUrl: Option[String] = Some(getFirstUrl(queryParams))
 
-    while (continueHarvest) getSinglePage(requestUrl) match {
+    while (requestUrl.isDefined) getSinglePage(requestUrl.get) match {
```

Comment:
If you remove the call to `.get` and pattern match on Some(x) etc, you'll get rid of a warning about the unsafe call to get.